### DISCUSSION TITLE
NUC Coral A+E in WiFi Slot

### DISCUSSION THREAD

**Discussion by frankyyy02** [blakeblackshear/frigate#2711] [Ask A Question]
With the long delays and unavailability of the Coral USB and M keys (at least in AUS), wondering if anybody has tried a Coral m.2 A+E key in an Intel NUC, replacing the WiFi card.
Have seen posts from Intel reps advising its dedicated to WiFi, and not other connectivity but unsure if this is a typical spec sheet response or a technical/BIOS limit?

Any experience with this? Alternatively, if you have a NUC-form factor like ASUS that has been successfully using m.2 A+E i'd be keen to explore it.

**Comment by blakeblackshear** [comment]
I have done this successfully in the Asus CN60 Chromebox. It's a PCIe slot, but a similar idea.

**Reply by frankyyy02** [reply]
Thanks @blakeblackshear 
Good to know...
Just checked again... it's now sold out too...
I'm seeing the USB version on backorder for up to twelve months... crazy.

**Comment by snipes040** [comment]
@blakeblackshear 

Just to confirm, the coral m.2 A+E key works successfully in a CN60 chromebox? I'm looking to get one since it's cheaper and has lower lead times than the USB version.

**Reply by blakeblackshear** [reply]
Yes. You need an adapter to put it in the mPCIe slot, but it works. That's what runs https://demo.frigate.video

**Comment by amzaldua** [comment]
Hi,
Are you received the Coral A+E? It works in wifi slot?

**Reply by kpcz** [reply]
I have https://www.asrock.com/mb/Intel/H470M-HDVM.2/index.asp MB, with dedicated M.2 Key E for wifi.  Coral m.2 A+E key works perfectly in WIFI slot. 
Before that I used wifi adapter https://www.amazon.com/Adapter-PCENGFF-NA1-Wireless-Network-Expansion/dp/B087RDYYBW also worked great.

**Comment by mtbBrian** [comment]
I made the same choice on a HP Slim Desktop S01-pF1025se. Dedicated wifi card removed from M.2 key slot and replaced with Coral. This worked without issue.

**Reply by amzaldua** [reply]
Both TPUs cores or just one of them?

**Reply by mtbBrian** [reply]
I have the M.2-2230-A-E-S3 board which only has one TPU core

**Comment by mtbBrian** [comment]
I ordered a Coral mini PCIe card at the same time as the M.2 version back in Feb 2022 from Mouser US. I got the M.2 version in July. Coincidently, I just got emailed that the PCIe shipped today. Haha.

**Reply by amzaldua** [reply]
Maybe I gave you luck. Let us know the results as soon as you test it. X)

**Comment by amzaldua** [comment]
Okay, I received the Coral TPU M.2 today and installed it in the WiFi slot. During the first boot, I lost LAN connectivity (WTF!?). So, I entered the BIOS and disabled the WiFi drivers to recover the Ethernet connectivity. Now, Frigate isn't detecting the Coral. What am I doing wrong?
I'm running the Frigate add-on on HA OS on a NUC11 with a Celeron processor.

**Reply by mtbBrian** [reply]
I roughly followed the official coral instructions here https://coral.ai/docs/m2/get-started/#requirements

Note that I installed in docker on a bare metal install of Ubuntu as Frigate docs recommend so there may be difference on your installation

**Reply by amzaldua** [reply]
Thank you mtbBrian, but I use the Home Assistant operating system. It is based on Debian, but I don't have access to make changes to the system at a deep level. In theory, the drivers should already be installed by default.

**Comment by amzaldua** [comment]
I'm completely lost on this issue. If I disable the Bluetooth and WiFi controllers in the BIOS, the HAOS starts correctly, but the Coral TPU is not detected. If I enable the Bluetooth and WiFi controllers in the BIOS setup, the HAOS starts, but I lose the Ethernet connection. When I enter the command "net info" in the CLI, this is the result:
![IMG_2894](https://user-images.githubusercontent.com/116205677/236648514-fb9d4190-2f75-4579-bc5d-b8a2476cc628.jpg)

**Comment by mtbBrian** [comment]
Hopefully you figured this out. If not, verify each of the BIOs settings are appropriate. There may be others as well not listed below that may be causing issues.

![image](https://github.com/blakeblackshear/frigate/assets/96792004/a99063a2-3757-40ac-8c3c-6c08e7182207)
![image](https://github.com/blakeblackshear/frigate/assets/96792004/e5abadc0-56cf-419a-86f1-609180bff062)
![image](https://github.com/blakeblackshear/frigate/assets/96792004/88fbe930-bd7a-4f85-a269-0819599e86f7)

**Reply by amzaldua** [reply]
Thanks a lot. I fixed this issue by reinstalling HAOS and restoring a full backup.

**Reply by froniusMW** [reply]
Having the exact same problem, will try an reinstall soon. Is the wifi disabled or enabled in the bios now that it works for you?
